### DISCUSSION TITLE
Search on keys and values

### DISCUSSION THREAD

**Discussion by LudwigNel** [vakata/jstree#2636] [Q&A]
Is it possible to search keys and values?

Looking at he following

<img width="384" alt="jsTree example" src="https://user-images.githubusercontent.com/14205687/176369614-c36298da-744b-477f-b53d-24240ea019c1.PNG">

I have a requirement to be able to search on the keys making up the tree nodes and their respective values.
As an example I have anode with a key about-us and a respective value About us. I need to be able to search for this node with a search of "about us"

I have implemented the ajax search where I return "about-us for that search but is looks like the client side still does the match on the keys only

Any suggestions on how I can get this to work?

**Comment by jplevene** [comment]
One way is to load the node data:
`var nodes = $("#tree").jstree("get_json", "#", {flat:true});`

Then loop through "nodes" to match your value:
```
$.each(nodes, function(i, node){
  if(node.data.some_field==="match") {
    // Do something
  }
});
```
You could use the Search plugin (https://www.jstree.com/plugins/), but this works on the node text, whereas above could be used to work on any node data field.

**Reply by LudwigNel** [reply]
Thanks for the suggestion.

I am currently using the search plugin, but as you mentioned is matches on node text.
And that is where my problem lies.

The above approach is not really going to help,but thanks.
I was hoping for a configuration changes or some way to override the client side search match
